lib/query_builder/schema/comparison_filter.ex

```diff
@@ -108,36 +108,72 @@ defmodule EctoShorts.QueryBuilder.Schema.ComparisonFilter do
     raise ArgumentError, message: "must provide a map for associations to filter on\ngiven #{inspect value}"
   end
 
-  defp build_relational_filter(query, binding_alias, filter_field, val) when is_list(val) do
+  defp build_relational_filter(query, binding_alias, filter_field, val, _relational_schema) when is_list(val) do
     where(query, [{^binding_alias, scm}], field(scm, ^filter_field) in ^val)
   end
 
-  defp build_relational_filter(query, binding_alias, filter_field, %NaiveDateTime{} = val) do
+  defp build_relational_filter(query, binding_alias, filter_field, %NaiveDateTime{} = val, _relational_schema) do
     where(query, [{^binding_alias, scm}], field(scm, ^filter_field) == ^val)
   end
 
-  defp build_relational_filter(query, binding_alias, filter_field, %DateTime{} = val) do
+  defp build_relational_filter(query, binding_alias, filter_field, %DateTime{} = val, _relational_schema) do
     where(query, [{^binding_alias, scm}], field(scm, ^filter_field) == ^val)
   end
 
-  defp build_relational_filter(query, binding_alias, field_key, filters) when is_map(filters) do
+  defp build_relational_filter(query, binding_alias, field_key, filters, relational_schema) when is_map(filters) do
+    cond do
+      field_key in relational_schema.__schema__(:query_fields) ->
+        build_relational_query_fields_filter(query, binding_alias, field_key, filters)
+      field_key in relational_schema.__schema__(:associations) ->
+        %{queryable: sub_relational_schema} = relational_schema.__schema__(:association, field_key)
+        build_relational_association_filter(query, binding_alias, field_key, filters, relational_schema, sub_relational_schema)
+      true ->
+        Logger.debug("[EctoShorts] #{Atom.to_string(field_key)} is neither a field nor has a valid association for #{relational_schema.__schema__(:source)} where filter")
+
+        query
+    end
+  end
+
+  defp build_relational_filter(query, binding_alias, filter_field, val, _relational_schema) do
+    where(query, [{^binding_alias, scm}], field(scm, ^filter_field) == ^val)
+  end
+
+  defp build_relational_query_fields_filter(query, binding_alias, field_key, filters) do
     Enum.reduce(filters, query, fn ({filter_type, value}, query_acc) ->
       build_relational_subfield_filter(query_acc, binding_alias, field_key, filter_type, value)
     end)
   end
 
-  defp build_relational_filter(query, binding_alias, filter_field, val) do
-    where(query, [{^binding_alias, scm}], field(scm, ^filter_field) == ^val)
+  defp build_relational_association_filter(_query, _binding_alias, field_key, _filters, relational_schema, nil) do
+    raise ArgumentError, message: "#{Atom.to_string(field_key)} does not have an associated schema for #{relational_schema.__schema__(:source)}"
+  end
+
+  defp build_relational_association_filter(query, binding_alias, field_key, filters, _relational_schema, sub_relational_schema) do
+    sub_binding_alias = :"#{binding_alias}_#{field_key}"
```

Comment:
Made an update to how this atom is built. It now references the original binding_alias and appends the new field_key onto it, so it will get larger the more you nest down

An example of what a query will look like
```
#Ecto.Query<from c0 in Schemas.Company, join: u1 in assoc(c0, :users),
 as: :ecto_shorts_users, join: p2 in assoc(u1, :posts),
 as: :ecto_shorts_users_posts, join: c3 in assoc(p2, :category),
 as: :ecto_shorts_users_posts_category, where: c3.name == ^"Television">
 ```